lib/omniauth/strategies/apple.rb

```diff
@@ -105,7 +123,7 @@ def verify_claims!(id_token)
         verify_aud!(id_token)
         verify_iat!(id_token)
         verify_exp!(id_token)
-        verify_nonce!(id_token) if id_token[:nonce_supported]
+        verify_nonce!(id_token) if id_token[:nonce_supported] && options[:nonce] != :ignore
```

Comment:
Out of curiosity, have you considered applying the suggestion made by @btalbot in https://github.com/nhosoya/omniauth-apple/issues/102 and just avoid verifying a blank `nonce` ?
```suggestion
        verify_nonce!(id_token) if id_token[:nonce_supported] && id_token[:nonce].present?
```

If so then may I ask why you decided not to use this solution ? It seems to work for me and requires a lot less changes in the code.

Reply:
Because this would pass even if `nonce` is required but nor returned, which I would consider a security risk.

Reply:
My bad, what I meant was to update `verify_nonce` like this
```ruby
      def verify_nonce!(id_token)
        invalid_claim! :nonce unless id_token[:nonce] == stored_nonce
      end
```

From my understanding `stored_nonce` will always be defined when the nonce is required and will not be defined when it's not so only checking that the token's nonce equals the stored_nonce would result in either checking whether "nonce is optional" or "nonce is required and it matches", wouldn't it ?

Reply:
@bvogel do you think that solution could work in the end ?

Reply:
No, as the `stored_nonce` is taken from the session that isn't present anymore if any `SameSite` strategy other then `:none` is used, the reason why I introduces the change in the first place